wsapi/wsapiV2.go

```diff
@@ -713,7 +712,6 @@ func HandleV2RawData(state interfaces.IState, params interface{}) (interface{},
 	hashkey := new(HashRequest)
 	err := MapToObject(params, hashkey)
 	if err != nil {
-		panic(reflect.TypeOf(params))
```

Comment:
Technically, this is a change in functionality, though it only affects API requests that send a non-string type for the "Hash" parameter. The change should be fine under the assumption that current clients send valid parameters.

Reply:
Is panicking really the right response to a bad type? Seems like it should return an invalid params. Is that what panicking does?

If panicking is the appropriate thing, the return statement should be removed as its dead code, which was confusing to me as it was not obvious that the panic wasn't just a remnant of someone's debugging process at one point.

Reply:
No, it's not. It's definitely leftover debug code that ideally shouldn't have been there. It's not a question of right vs wrong, but "changing functionality". Removing bugs can sometimes break programs that depend on those bugs to work.

**I don't think this is one of those cases and the change is fine.**



> Is panicking really the right response to a bad type? Seems like it should return an invalid params. Is that what panicking does?

It's not. panicking will kill the http thread and produce the panic in the console (without impacting the node at all). The client will get no output at all.

I tested it just to make sure and a bad request on the server side looks like this:
```
2020/02/13 08:32:03 http: panic serving x.x.x:57191: map[string]interface {}
goroutine 781658 [running]:
net/http.(*conn).serve.func1(0xc0032d0640)
        /usr/local/go/src/net/http/server.go:1769 +0x139
panic(0xdc2920, 0xccf6e0)
        /usr/local/go/src/runtime/panic.go:522 +0x1b5
github.com/FactomProject/factomd/wsapi.HandleV2RawData(0x1089640, 0xc00016e000, 0xccf6e0, 0xc00b24fce0, 0x0, 0x0, 0x0)
        /go/src/github.com/FactomProject/factomd/wsapi/wsapiV2.go:677 +0x92d
github.com/FactomProject/factomd/wsapi.HandleV2JSONRequest(0x1089640, 0xc00016e000, 0xc002ea49c0, 0x0, 0x0)
        /go/src/github.com/FactomProject/factomd/wsapi/wsapiV2.go:131 +0x676
github.com/FactomProject/factomd/wsapi.HandleV2(0x10644a0, 0xc0005ac1c0, 0xc023c47300)
        /go/src/github.com/FactomProject/factomd/wsapi/wsapiV2.go:65 +0x3dc
github.com/FactomProject/factomd/wsapi.APILogger.func1.1(0x10644a0, 0xc0005ac1c0, 0xc023c47300)
        /go/src/github.com/FactomProject/factomd/wsapi/server.go:96 +0x16d
github.com/FactomProject/factomd/wsapi.IDInjector.func1.1(0x10644a0, 0xc0005ac1c0, 0xc023c47300)
        /go/src/github.com/FactomProject/factomd/wsapi/server.go:106 +0x136
net/http.HandlerFunc.ServeHTTP(0xc00226c3e0, 0x10644a0, 0xc0005ac1c0, 0xc023c47300)
        /usr/local/go/src/net/http/server.go:1995 +0x44
github.com/FactomProject/factomd/vendor/github.com/gorilla/mux.(*Router).ServeHTTP(0xc001c940c0, 0x10644a0, 0xc0005ac1c0, 0xc023c47100)
        /go/src/github.com/FactomProject/factomd/vendor/github.com/gorilla/mux/mux.go:210 +0xe3
net/http.serverHandler.ServeHTTP(0xc003f604e0, 0x10644a0, 0xc0005ac1c0, 0xc023c47100)
        /usr/local/go/src/net/http/server.go:2774 +0xa8
net/http.(*conn).serve(0xc0032d0640, 0x1067760, 0xc002ea4880)
        /usr/local/go/src/net/http/server.go:1878 +0x851
created by net/http.(*Server).Serve
        /usr/local/go/src/net/http/server.go:2884 +0x2f4
```

and on the client side:
```
curl --data-binary "{\"jsonrpc\": \"2.0\", \"id\": 0, \"method\": \"raw-data\", \"params\": {\"hash\": 1}}" -H "content-type:text/plain;" http://x.x.x:8088/v2
curl: (52) Empty reply from server
```

Reply:
Thanks for the thorough response.